src/main/java/io/oneko/docker/DockerRegistryPolling.java

```diff
@@ -55,6 +57,9 @@
 	private final AtomicBoolean pollingInProgress = new AtomicBoolean(false);
 	private final ExpiringBucket<UUID> failedManifestRequests = new ExpiringBucket<UUID>(Duration.ofMinutes(5)).concurrent();
 
+	private Instant lastPollingJobStartDate = null;
```

Comment:
Since this is reset always, it should be namend currentPollingJobStartDate.
I even think, that It might be helpful to gather date and disposable into a DockerRegistryPollingJob (or something like that) class, that support control methods like shouldCancel() cancel etc.

Reply:
And such a class might be all covered in unit tests. Wouldn't that be fun?

Reply:
> Wouldn't that be fun?

Indeed!